Tests/SwiftDocCTests/Diagnostics/DiagnosticConsoleWriterTests.swift

```diff
@@ -127,28 +127,329 @@ class DiagnosticConsoleWriterTests: XCTestCase {
-        let problem = Problem(diagnostic: diagnostic, possibleSolutions: [solution])
-        
+        let expectedPath = "--> /path/to/file.md:1:8-10:21"
+
+        do {
+            let logger = Logger()
+            let consumer = DiagnosticConsoleWriter(logger, highlight: true)
+            let diagnostic = Diagnostic(source: source, severity: .error, range: range, identifier: identifier, summary: summary, explanation: explanation)
+            let problem = Problem(diagnostic: diagnostic, possibleSolutions: [])
+            consumer.receive([problem])
+            try? consumer.finalize()
+            XCTAssertEqual(logger.output, """
+            \u{001B}[1;31merror: \(summary)\u{001B}[0;39m
+            \(explanation)
+            \(expectedPath)
+            """)
+        }
+
+        do {
+            let logger = Logger()
+            let consumer = DiagnosticConsoleWriter(logger, highlight: true)
+            let diagnostic = Diagnostic(source: source, severity: .warning, range: range, identifier: identifier, summary: summary, explanation: explanation)
+            let problem = Problem(diagnostic: diagnostic, possibleSolutions: [])
+            consumer.receive([problem])
+            try? consumer.finalize()
+            XCTAssertEqual(logger.output, """
+            \u{001B}[1;33mwarning: \(summary)\u{001B}[0;39m
+            \(explanation)
+            \(expectedPath)
+            """)
+        }
+
+        do {
+            let logger = Logger()
+            let consumer = DiagnosticConsoleWriter(logger, highlight: true)
+            let diagnostic = Diagnostic(source: source, severity: .hint, range: range, identifier: identifier, summary: summary, explanation: explanation)
+            let problem = Problem(diagnostic: diagnostic, possibleSolutions: [])
+            consumer.receive([problem])
+            try? consumer.finalize()
+            XCTAssertEqual(logger.output, """
+            \u{001B}[1;39mnotice: \(summary)\u{001B}[0;39m
+            \(explanation)
+            \(expectedPath)
+            """)
+        }
+
+        do {
+            let logger = Logger()
+            let consumer = DiagnosticConsoleWriter(logger, highlight: true)
+            let diagnostic = Diagnostic(source: source, severity: .information, range: range, identifier: identifier, summary: summary, explanation: explanation)
+            let problem = Problem(diagnostic: diagnostic, possibleSolutions: [])
+            consumer.receive([problem])
+            try? consumer.finalize()
+            XCTAssertEqual(logger.output, """
+            \u{001B}[1;39mnote: \(summary)\u{001B}[0;39m
+            \(explanation)
+            \(expectedPath)
+            """)
+        }
+    }
+
+    func testDefaultFormatting_DisplaysRelativePath() {
+        let baseURL = URL(string: "/path/to")!
+        let source = URL(string: "/path/to/file.md")!
```

Comment:
minor: I see that the existing tests also use `URL(string:)` but if these used `URL(fileURLWithPath:)` they wouldn't need to force unwrap the result